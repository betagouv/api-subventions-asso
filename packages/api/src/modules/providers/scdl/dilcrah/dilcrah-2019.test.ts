import path from "path";
import fs from "fs";
import ScdlGrantParser from "../scdl.grant.parser";
import { DILCRAH_2019_MAPPER } from "./dilcrah-2019.mapper";

describe("TEST DILCRAH IMPORT", () => {
    it("should extract data", () => {
        const csvBuffer = fs.readFileSync(path.resolve(__dirname, "../__fixtures__/DILCRAH_2019.csv"));
        // @ts-expect-error: give fake ObjectId
        const actual = ScdlGrantParser.parseCsv(csvBuffer, DILCRAH_2019_MAPPER, "652527d5d86905f43496e67b");
        expect(actual).toMatchSnapshot();
    });
});
