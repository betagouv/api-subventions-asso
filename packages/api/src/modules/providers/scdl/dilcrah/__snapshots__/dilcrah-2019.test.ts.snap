// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST DILCRAH IMPORT should extract data 1`] = `
Array [
  Object {
    "UeNotification": false,
    "allocatorName": "SPM - DILCRAH",
    "allocatorSiret": "12000123400023",
    "amount": 45000,
    "associationName": "ANNO_1",
    "associationSiret": "13579246800043",
    "conventionDate": 2019-10-04T22:00:00.000Z,
    "decisionReference": "2102661732",
    "extractId": "652527d5d86905f43496e67b",
    "grantPercentage": 1,
    "idRAE": "Non",
    "object": "Immigration et territoire de la politique de la ville - citoyenneté et héros militaires - Lutte contre les discriminations, le racisme et l'antisémitisme - Figures de réussite, figures de l'exemplarité républicaine : artistes",
    "paymentConditions": "UNIQUE",
    "paymentEndDate": 2001-04-30T22:00:00.000Z,
    "paymentStartDate": Date { NaN },
  },
  Object {
    "UeNotification": false,
    "allocatorName": "SPM - DILCRAH",
    "allocatorSiret": "12000123400023",
    "amount": 50000,
    "associationName": "ANNO_2",
    "associationSiret": "11223344500034",
    "conventionDate": Date { NaN },
    "decisionReference": "2102763867",
    "extractId": "652527d5d86905f43496e67b",
    "grantPercentage": 1,
    "idRAE": "Non",
    "object": "Les engagés de l'AFEV contre le racisme et l'antisémistisme",
    "paymentConditions": "UNIQUE",
    "paymentEndDate": 2001-08-31T22:00:00.000Z,
    "paymentStartDate": 2001-03-31T22:00:00.000Z,
  },
  Object {
    "UeNotification": false,
    "allocatorName": "SPM - DILCRAH",
    "allocatorSiret": "12000123400023",
    "amount": 5000,
    "associationName": "ANNO_1",
    "associationSiret": "98876554100014",
    "conventionDate": Date { NaN },
    "decisionReference": "2102672573",
    "extractId": "652527d5d86905f43496e67b",
    "grantPercentage": 1,
    "idRAE": "Non",
    "object": "Pride Marseille 2019",
    "paymentConditions": "UNIQUE",
    "paymentEndDate": 2001-04-30T22:00:00.000Z,
    "paymentStartDate": Date { NaN },
  },
  Object {
    "UeNotification": false,
    "allocatorName": "SPM - DILCRAH",
    "allocatorSiret": "12000123400023",
    "amount": 10000,
    "associationName": "ANNO_3",
    "associationSiret": "74185296300016",
    "conventionDate": Date { NaN },
    "decisionReference": "2102804885",
    "extractId": "652527d5d86905f43496e67b",
    "grantPercentage": 1,
    "idRAE": "Non",
    "object": "\\"\\"\\"A la mémoire des travailleurs chinois de la Grande guerre et du traité de Versailles\\"\\"\\"",
    "paymentConditions": "UNIQUE",
    "paymentEndDate": 2001-09-30T22:00:00.000Z,
    "paymentStartDate": Date { NaN },
  },
]
`;
